src/collective/documentgenerator/browser/generation_view.py

```diff
@@ -157,6 +158,10 @@ def _render_document(self, pod_template, output_format, sub_documents, **kwargs)
         helper_view = self.get_generation_context_helper()
         generation_context = self._get_generation_context(helper_view, pod_template=pod_template)
         # enrich the generation context with previously generated documents
+        # if they do not overwrite the current context variables
+        utils.validate_dict_update(generation_context, sub_documents,
+                                   "Duplicate context variable found. "
```

Comment:
not context variable here! => _("Error when merging merge_templates in generation context for key")